nilearn/plotting/surface/tests/test_backend.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+
+from nilearn._utils.helpers import is_matplotlib_installed, is_plotly_installed
+from nilearn.plotting.surface._backend import get_surface_backend
+
+
+@pytest.mark.skipif(
+    is_matplotlib_installed(),
+    reason="This test is run only if matplotlib is not installed.",
+)
+def test_get_surface_backend_matplotlib_not_installed():
+    """Tests to see if get_surface_backend raises error when matplotlib is not
+    installed.
+    """
+    with pytest.raises(ImportError, match="Using engine"):
+        get_surface_backend("matplotlib")
+
+
+@pytest.mark.skipif(
+    is_plotly_installed(),
+    reason="This test is run only if plotly is not installed.",
+)
+def test_get_surface_backend_plotly_not_installed():
+    """Tests to see if get_surface_backend raises error when plotly is not
+    installed.
+    """
+    with pytest.raises(ImportError, match="Using engine"):
+        get_surface_backend("plotly")
+
+
+def test_get_surface_backend_unknown_error():
+    """Tests to see if get_surface_backend raises error when plotly is not
+    installed.
+    """
+    with pytest.raises(ValueError, match="Unknown plotting"):
+        get_surface_backend("unknown")
+
+
+class BaseTestSurfaceBackend:
```

Comment:
Why do we need a class here ?

Reply:
I created this base test class, that has all necessary methods to test both backends. 

When I extend this in `test_plotly_backend.py` it will take `engine='plotly'` and when I extend it in `test_matplotlib_backend.py`, it will take `engine='matplotlib'` and run all tests in the base class. I do not need to rewrite  the same tests for both backends, or check with if conditions depending on the engine, etc. 

Tests in the backend test files won't run, unless the backend is installed.